src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -407,6 +405,16 @@ internal static Collection<CustomAttributeBuilder> CreateTypeAttributes(ScriptHo
             return customAttributes;
         }
 
+        private void RestartHost()
+        {
+            // signal host restart
+            _restartEvent.Set();
+
+            // whenever we're restarting the host, we want to let the Node
+            // invoker know so it can clear the require cache, etc.
+            NodeFunctionInvoker.OnHostRestart();
```

Comment:
Yes I agree, and I thought about this before putting this here. Note that this is an existing issue - the ctor also references NodeFunctionInvoker directly. I've logged https://github.com/Azure/azure-webjobs-sdk-script/issues/959 for this and will address this in a separate PR.